client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import deepFreeze from 'deep-freeze';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import { merge } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { useFakeTimers } from 'test/helpers/use-sinon';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { retryOnFailure as rof } from '../';
+import { noRetry, exponentialBackoff } from '../policies';
+
+const retryOnFailure = rof();
+const retryWithDelay = delay => rof( () => delay );
+
+const nextError = { fail: 'failed big time' };
+
+const getSites = deepFreeze( http( {
+	method: 'GET',
+	path: '/sites',
+	apiVersion: 'v1',
+} ) );
+
+const withRetries = retryCount => actionOrInbound =>
```

Comment:
this is a helper for tests only so that we didn't need two functions for the same basic notion. it wouldn't be that big of a deal but I felt when I wrote it that being able to simply indicate "this thing has already had N retry attempts" was justifiable for the function-overload (it's not a complicated function at least)

in the first case we augment an HTTP object for the pipeline to say that it is coming in after N retry attempts
in the second case we are augmenting an action for the same effect